packages/app/src/store/run-all-specs-store.ts

```diff
@@ -0,0 +1,96 @@
+import { RUN_ALL_SPECS_KEY } from '@packages/types/src'
+import { gql, useMutation, useQuery } from '@urql/vue'
+import { defineStore } from 'pinia'
+import { computed, ref } from 'vue'
+import { useRouter } from 'vue-router'
+import { RunAllSpecsDataDocument, RunAllSpecsDocument } from '../generated/graphql-test'
+import { getSeparator, SpecTreeNode, UseCollapsibleTreeNode } from '../specs/tree/useCollapsibleTree'
+
+type ResolvedConfig = { value: any, from: 'string', field: string }[]
+
+gql`
+query RunAllSpecsData {
+  currentProject {
+    id
+    config
+    currentTestingType
+  }
+}
+`
+
+gql`
+mutation RunAllSpecs ($specPath: String!, $runAllSpecs: [String!]!) {
+  setRunAllSpecs(runAllSpecs: $runAllSpecs)
+  launchOpenProject(specPath: $specPath) {
+    id
+  }
+}
+`
+
+const isRunMode = window.__CYPRESS_MODE__ === 'run' && window.top === window
```

Comment:
We've got something like

```js
const isRunMode = window.__CYPRESS_MODE__ === 'run'
```

In quite a few places - any chance we can move this to a single function?

Reply:
Can do, I think the `window.top === window` is for cy-in-cy? Makes sense to consolidate that as well, going to include that and see if anything breaks.